sceptre/config/strategies.py

```diff
@@ -21,16 +22,17 @@ def list_join(a, b):
     """
     if a and not isinstance(a, list):
         raise TypeError('{} is not a list'.format(a))
+
     if b and not isinstance(b, list):
         raise TypeError('{} is not a list'.format(b))
 
     if a is None:
-        return b
+        return deepcopy(b)
```

Comment:
This change helps prevent mutable values that are shared across a stack group (like the dependencies list) from being the _same_ object in memory across stacks. This prevents a weird bug that can happen when one stack's value gets modified by another stack.